Automattic/wp-calypso
# Jetpack Cloud: update copy in the "Fix all threats" dialog

#### Changes proposed in this Pull Request

* Update copy that we show inside the "Fix all threats" dialog. The goal is to list the actions we would do to fix the threats.
* Added a [Cancel] button. Before, there was no way to close the dialog.
* Created a utils file to encapsulate functionality related to threat objects.

#### Implementation notes

* Added specific types for threats that can be fixed/ignored. TS is happy about this.

#### Questions

* Should we display the [Fix all] button when there is only one fixable threat? I tend to think we should not, but what if the user has a lot of "ignorable" threats and just one "fixable". That "fixable" threat might get lost in the list and the [Fix all] button could help the user see there is a threat.

#### Testing instructions

* Run this PR
* Select a site with as many threats as possible (hopefully, you will have threats of different types)
* Go to Scan -> Scanner
* Click the [Fix all] button
* Verify the copy makes sense (plural)
* Verify you see a short description of the threat and the action Jetpack is going to do to fix the threat

#### Before

![image](https://user-images.githubusercontent.com/3418513/80417745-9ee85480-88ac-11ea-8b3d-49a43985e7bf.png)

#### After

![image](https://user-images.githubusercontent.com/3418513/80528455-53957b00-896c-11ea-9fa9-1c173dbe98c7.png)
